pkg/server/status/recorder.go

```diff
@@ -888,6 +983,64 @@ func (rr registryRecorder) recordChild(
 	})
 }
 
+// recordChildx filters the metrics in the registry down to those provided in
+// the metricsFilter argument, and iterates through any child metrics that
+// may exist on said metric. Records up to 1000 child metrics to prevent
+// unbounded memory usage and performance issues.
+//
+// NB: Only available for Counter and Gauge metrics.
+func (rr registryRecorder) recordChildx(
+	dest *[]tspb.TimeSeriesData,
+) {
+	const maxChildMetrics = 1000
+	var childMetricsCount int
+
+	labels := rr.registry.GetLabels()
+	rr.registry.Each(func(name string, v interface{}) {
+		prom, ok := v.(metric.PrometheusExportable)
+		if !ok {
+			return
+		}
+		promIter, ok := v.(metric.PrometheusIterable)
+		if !ok {
+			return
+		}
+		m := prom.ToPrometheusMetric()
+		m.Label = append(labels, prom.GetLabels(false /* useStaticLabels */)...)
+
+		processChildMetric := func(metric *prometheusgo.Metric) {
+			if childMetricsCount >= maxChildMetrics {
+				return // Stop processing once we hit the limit
+			}
+
+			childmetricname := prom.GetName(false /* useStaticLabels */)
+			for _, label := range metric.Label {
+				childmetricname = fmt.Sprintf("%s___%s_%q_jason", childmetricname, label.GetName(), label.GetValue())
+			}
```

Comment:
you'll probably want to use a string builder here so that you construct the final string just once